prometheus-wireguard-exporter/riscv64.patch

```diff
@@ -0,0 +1,32 @@
+diff --git repos/community-x86_64/PKGBUILD repos/community-x86_64/PKGBUILD
+index a573d5d..2528e6d 100644
+--- repos/community-x86_64/PKGBUILD
++++ repos/community-x86_64/PKGBUILD
+@@ -23,7 +23,26 @@ sha512sums=('9ff42deaf28a45be26cf0211b11878be873467bee0bdb128efec3c1a9d4f7214f53
+ 
+ prepare() {
+   cd "prometheus_wireguard_exporter-${pkgver}"
+-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
++
++  # RISC-V Patches:
++  # rustc(1) recognizes `riscv64gc' instead of `riscv64'. Thus,
++  # we need to manually substitute riscv64 with riscv64gc in 
++  # --target. The `--target' argument, according to Arch Wiki,
++  # "tells cargo to only fetch dependencies needed for the
++  # specific target platform being built, thus reducing downloads."
++  # Ref: https://wiki.archlinux.org/title/Rust_package_guidelines#Prepare
++  local _target="$CARCH-unknown-linux-gnu"
++  if test $CARCH = "riscv64"; then
++    _target="${CARCH}gc-unknown-linux-gnu"
++
++    # The upstream `ring' package does not support riscv yet.
++    # This workaround comes from https://github.com/felixonmars/
++    # archriscv-packages/wiki/%E6%88%91%E4%BB%AC%E7%9A%84%E5%B7
++    # %A5%E4%BD%9C%E4%B9%A0%E6%83%AF
++    echo -e "\n[patch.crates-io]\nring = { git = 'https://github.com/felixonmars/ring', branch = '0.16.20' }" >> Cargo.toml
++    cargo update -p ring
++  fi
++  cargo fetch --locked --target $_target
```

Comment:
```suggestion
+  cargo fetch --locked
```
We can just do like this as a convention. 😄

Reply:
Thanks!